src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -215,61 +230,88 @@ private UUID reserveEntityKeyId(
     }
 
     private UUID reserveDeviceEntityKeyId(
+            UUID devicesESID,
             Map<UUID, Set<Object>> data,
             DataIntegrationApi dataIntegrationApi ) {
 
         return reserveEntityKeyId(
-                entitySetIdMap.get( DEVICES_ENTITY_SET_NAME ),
+                devicesESID,
                 ImmutableList.of( propertyTypeIdsByFQN.get( STRING_ID_FQN ) ),
                 data,
                 dataIntegrationApi
         );
     }
 
-    private UUID getStudyEntityKeyId( UUID studyId ) {
-        logger.info( "Retrieving studyEntityKeyId, studyId = {}", studyId );
-        if ( studies.containsKey( studyId ) ) {
-            return studies.get( studyId );
-        }
+    private UUID getStudyEntityKeyId( UUID organizationId, UUID studyId ) {
+        logger.info( "Retrieving studyEKID: studyId = {}, orgId = {}", studyId, organizationId );
+        try {
+            // api
+            ApiClient apiClient = prodApiClientCache.get( ApiClient.class );
+            DataApi dataApi = apiClient.getDataApi();
 
-        logger.error( "Failed to retrieve studyEntityKeyId, studyId = {}", studyId );
-        return null;
+            UUID entitySetId = ensureEntitySetExists( organizationId, CHRONICLE_CORE, STUDIES );
+
+            String jwtToken = auth0Client.getIdToken( username, password );
+
+            Iterable<SetMultimap<FullQualifiedName, Object>> data = dataApi
+                    .loadEntitySetData( entitySetId, FileType.json, jwtToken );
+
+            return StreamUtil.stream( data )
+                    .filter( entry -> !entry.get( STRING_ID_FQN ).isEmpty() )
+                    .filter( entry -> entry.get( STRING_ID_FQN ).iterator().next().toString()
+                            .equals( studyId.toString() ) )
+                    .map( entry -> UUID.fromString( entry.get( ID_FQN ).iterator().next().toString() ) )
+                    .findFirst().orElse( null );
+
+        } catch ( Exception e ) {
+            String error = "failed to get EKID of study with studyId " + studyId + " in organization " + organizationId;
+            logger.error( error, e );
```

Comment:
I think it's much easier to use the log4j syntax here instead of constructing a separate string, i.e.
```
logger.error( "Failed to get EKID of study with studyId {} in organization {}", studyId, organizationId, e );
```